java-sdk/src/main/java/org/sagebionetworks/bridge/sdk/UserSession.java

```diff
@@ -3,43 +3,53 @@
 import static org.sagebionetworks.bridge.sdk.utils.Utilities.TO_STRING_STYLE;
 
 import java.util.Map;
-import java.util.Set;
 
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
+import org.sagebionetworks.bridge.sdk.exceptions.BridgeSDKException;
 import org.sagebionetworks.bridge.sdk.json.SubpopulationGuidKeyDeserializer;
+import org.sagebionetworks.bridge.sdk.models.accounts.StudyParticipant;
 import org.sagebionetworks.bridge.sdk.models.subpopulations.ConsentStatus;
 import org.sagebionetworks.bridge.sdk.models.subpopulations.SubpopulationGuid;
-import org.sagebionetworks.bridge.sdk.models.users.SharingScope;
+import org.sagebionetworks.bridge.sdk.utils.Utilities;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.google.common.collect.ImmutableMap;
 
 final class UserSession {
 
+    private static final TypeReference<Map<SubpopulationGuid, ConsentStatus>> CONSENT_STATUS_MAP = new TypeReference<Map<SubpopulationGuid, ConsentStatus>>() {};
+    private static final ObjectMapper MAPPER = Utilities.getMapper();
+    
+    public static final UserSession fromJSON(JsonNode node) {
+        try {
+            String sessionToken = node.get("sessionToken").asText();
+            boolean authenticated = node.get("authenticated").asBoolean();
+            Map<SubpopulationGuid, ConsentStatus> consentStatuses = MAPPER.convertValue(
+                    node.get("consentStatuses"), CONSENT_STATUS_MAP);
+            StudyParticipant studyParticipant = MAPPER.treeToValue(node, StudyParticipant.class);
+            return new UserSession(sessionToken, authenticated, consentStatuses, studyParticipant);
+        } catch(JsonProcessingException e) {
+            throw new BridgeSDKException(e.getMessage(), e);
+        }
+    }
```

Comment:
The study participant properties are collapsed into the session object JSON for backwards compatibility with the existing session object (when serializing), and then pulled back out into the StudyParticipant object when deserializing (hence things like dataGroups are removed in favor of accessing them via the StudyParticipant). I didn't see a way to do this using Jackson's annotations.